src/main/java/sda/groupProject/knowledgeChecker/data/JSONConnector.java

```diff
@@ -104,4 +105,80 @@ public ArrayList<Question> getListOfQuestions(Advancement level, Category catego
                 .collect(Collectors.toList());
     }
 
+    public ArrayList<Question> getListOfQuestions(Advancement level, List<String> categoryNames) {
+        return (ArrayList<Question>) questionArrayList
+                .stream()
+                .filter(question -> question.advancement().equals(level))
+                .filter(question -> categoryNames.contains(question.category().categoryName()))
+                .collect(Collectors.toList());
+    }
+
+    public ArrayList<Question> getListOfQuestions(Advancement level, List<String> categoryNames,
+                                                  boolean randomSelection) {
+        List<Question> filteredQuestions = questionArrayList
+                .stream()
+                .filter(question -> question.advancement().equals(level))
+                .filter(question -> categoryNames.contains(question.category().categoryName()))
+                .collect(Collectors.toList());
+
+        if (randomSelection) {
+            Collections.shuffle(filteredQuestions);
+        }
+
+        return new ArrayList<>(filteredQuestions);
+    }
+
+    public ArrayList<Question> getListOfQuestions(Advancement level, List<String> categoryNames,
+                                                  boolean randomSelection, int numberOfQuestions) {
+        List<Question> filteredQuestions = questionArrayList
+                .stream()
+                .filter(question -> question.advancement().equals(level))
+                .filter(question -> categoryNames.contains(question.category().categoryName()))
+                .collect(Collectors.toList());
+
+        if (randomSelection) {
+            Collections.shuffle(filteredQuestions);
+        }
+
+        if (numberOfQuestions >= filteredQuestions.size()) {
```

Comment:
Excellent! It is what I've wanted.
But now I think that it can be reduced.
Please make one method:
categoryNames.contains(question.category().categoryName()))
always shuffled
always numberOfQuestions (you can do .limit in stream)

Write
Preview
Attach files by dragging & dropping, selecting or pasting them.
Cancel
Update comment